packages/legend-query-builder/src/stores/QueryBuilderConfig.ts

```diff
@@ -25,12 +25,13 @@ export const DEFAULT_VARIABLE_NAME = 'var';
 export const DEFAULT_CONSTANT_VARIABLE_NAME = 'c_var';
 export const DEFAULT_POST_FILTER_LAMBDA_VARIABLE_NAME = 'row';
 
-export const QUERY_BUILDER_PROPERTY_SEARCH_MAX_DEPTH = 5;
+export const QUERY_BUILDER_PROPERTY_SEARCH_MAX_DEPTH = 10;
```

Comment:
Since we're eliminating circular dependencies now, it should be safe to increase this depth.